src/backend/api/Fusion.Resources.Domain/Queries/Departments/GetRelevantOrgUnits.cs

```diff
@@ -0,0 +1,382 @@
+﻿using Fusion.AspNetCore.OData;
+using Fusion.Integration;
+using Fusion.Integration.LineOrg;
+using Fusion.Integration.LineOrg.Cache;
+using Fusion.Integration.Profile;
+using Fusion.Integration.Roles;
+using Fusion.Resources.Domain.Models;
+using Fusion.Services.LineOrg.ApiModels;
+using MediatR;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+using System;
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+
+namespace Fusion.Resources.Domain.Queries
+{
+    /// <summary>
+    /// Fetch the profile for a resource owner. Will compile a list of departments the person has responsibilities in and which is relevant.
+    /// </summary>
+    public class GetRelevantOrgUnits : IRequest<QueryRangedList<QueryRelevantOrgUnit>>
+    {
+        public GetRelevantOrgUnits(string profileId, AspNetCore.OData.ODataQueryParams query)
+        {
+            ProfileId = profileId;
+            Query = query;
+        }
+
+        /// <summary>
+        /// Mail or azure unique id
+        /// </summary>
+        public PersonId ProfileId { get; set; }
+        public ODataQueryParams Query { get; }
+        public const string CACHEKEY = "Cache.OrgUnits";
+        public enum Roles
+        {
+            DelegatedManager,
+            DelegatedParentManager,
+            ParentManager,
+            Manager,
+            Write
+        }
+
+        public class Handler : IRequestHandler<GetRelevantOrgUnits, QueryRangedList<QueryRelevantOrgUnit>>
+        {
+            private readonly TimeSpan defaultAbsoluteCacheExpirationHours = TimeSpan.FromHours(1);
+            private readonly ILogger<Handler> logger;
+            private readonly IFusionProfileResolver profileResolver;
+            private readonly IFusionRolesClient rolesClient;
+            private readonly IMediator mediator;
+            private readonly ILineOrgResolver lineOrgResolver;
+            private readonly IMemoryCache memCache;
+            private readonly HttpClient lineOrgClient;
+
+
+
+
+            public Handler(ILogger<Handler> logger, IFusionProfileResolver profileResolver, IFusionRolesClient rolesClient, ILineOrgResolver lineOrgResolver, IMediator mediator, IMemoryCache memCache, IHttpClientFactory httpClientFactory)
+            {
+                this.logger = logger;
+                this.profileResolver = profileResolver;
+                this.rolesClient = rolesClient;
+                this.mediator = mediator;
+                this.lineOrgResolver = lineOrgResolver;
+                this.memCache = memCache;
+                this.lineOrgClient = httpClientFactory.CreateClient(IntegrationConfig.HttpClients.ApplicationLineOrg());
+
+
+
+
+
+            }
+
+            public async Task<List<QueryRelevantOrgUnit>> ResolveAllOrgUnits()
```

Comment:
I would suggest GetAllOrgUnitsAsync here.

-Async since it is async.
Get- since we know and expect data to be retrieved.